### PR TITLE
Add SIP-78 - iSynth limit reset must not trip circuit breaker

### PR DESCRIPTION
**Pull request by justinjmoses**
Ensure the repricing of an iSynth does not trigger the decentralized circuit breaker.

### CODE REVIEW

File: SIPS/sip-78.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+<!--"If you can't explain it simply, you don't understand it well enough." Simply describe the outcome the proposed changes intends to achieve. This should be non-technical and accessible to a casual community member.-->
+
+Ensure the repricing of an iSynth does not trigger the decentralized circuit breaker.
+
+## Abstract
+
+<!--A short (~200 word) description of the proposed change, the abstract should clearly describe the proposed change. This is what *will* be done if the SIP is implemented, not *why* it should be done or *how* it will be done. If the SIP proposes deploying a new contract, write, "we propose to deploy a new contract that will do x".-->
+
+When the protocolDAO resets an iSynth, it must also reset the pricing for the last exchange price tracker in the `Exchanger` contract.
```
Comment (jacko125) [inline SIPS/sip-78.md:23]: Should it reset the pricing to the new `EntryPoint` of the iSynth so that it "starts" on the same level (as discussed).

Comment (justinjmoses) [reply SIPS/sip-78.md:23]: yes- let me rephrase it.

File: SIPS/sip-78.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+N/A
+
+### Test Cases
+
+<!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation..-->
+
+- Given `iETH` is an inverse synth tracking `ETH`
+- And there exists a previous exchange into or out of `iETH`, ensuring `Exchanger.lastExchangeRate(iETH)` has an entry
+- When the protocolDAO invokes setInversePricing, changing the rate more than the `Exchanger.priceDeviationThresholdFactor`
```
Comment (jacko125) [inline SIPS/sip-78.md:65]: Resetting / deleting `lastExchangeRate(iETH)` to 0 would technically work here too 

```
        uint lastRateFromExchange = lastExchangeRate[currencyKey];

        if (lastRateFromExchange > 0) {
            return _isDeviationAboveThreshold(lastRateFromExchange, currentRate);
        }
```

**Review by jacko125 — APPROVED** [review]
lgtm, had two comments in approach to this change.

File: SIPS/sip-78.md (hunk 2 above)
Comment (justinjmoses) [reply SIPS/sip-78.md:65]: Actually it would instead then look up the last 3 prices on-chain, which after SIP-75 in Pollux, would mean calculating the iSynth from previous long rates. This would still work in most cases, but isn't as accurate as setting an explicit `lastRateFromExchange` for the iSynth

Comment (jacko125) [reply SIPS/sip-78.md:65]: @justinjmoses  Yeah , should be setting it to the new `EntryPoint` of the iSynths that it's getting resetted to. So the system will acknowledge last trade price is new 'EntryPoint'

Comment (justinjmoses) [reply SIPS/sip-78.md:65]: well it will set it to the new iSynth price calculated from the current long rate and the new entryPoint. That's what you meant right?
